main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
+	// create a "pid" directory under cache directory and clean up inactive pid file
+	pidPath := filepath.Join(c.Dir, "pid")
+	err := os.MkdirAll(pidPath, os.FileMode(0744))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = filepath.Walk(pidPath, func(name string, fileInfo os.FileInfo, err error) error {
+		if !fileInfo.IsDir() {
+			pid, _ := strconv.Atoi(fileInfo.Name())
+			bazelRemoteProcess, err := os.FindProcess(pid)
```

Comment:
If `strconv.AtoI()` fails `pid` can be null.

Reply:
strconv.AtoI() always returns an int

Reply:
Sorry, I should've been more clear. If there is a file that can't be converted to a number, for example a file called "junk", then `strong.Atoi()` will return 0, which is a valid argument to `os.FindProcess()` as it'll just wrap it in a `*Process`. Below, in the call to `Signal`, it'll use PID 0 which means send the signal to all the processes with the same group id as the sender, effectively killing the sender itself.